tests/Tests/500__jupyterhub/culler.robot

```diff
@@ -0,0 +1,169 @@
+*** Settings ***
+Documentation    Tests for the NB culler
+Resource         ../../Resources/ODS.robot
+Resource         ../../Resources/Common.robot
+Resource         ../../Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
+Resource         ../../Resources/Page/ODH/JupyterHub/JupyterLabLauncher.robot
+Resource         ../../Resources/Page/ODH/ODHDashboard/ODHDashboard.robot
+Library          ../../../libs/Helpers.py
+Library          OpenShiftLibrary
+Suite Teardown   Teardown
+Force Tags       JupyterHub
+
+
+*** Variables ***
+${DEFAULT_CULLER_TIMEOUT} =    31536000
+${CUSTOM_CULLER_TIMEOUT} =     600
+
+
+*** Test Cases ***
+Verify Default Culler Timeout
+    [Documentation]    Checks default culler timeout
+    [Tags]    Sanity    Tier1
+    ...       ODS-1255
+    Disable Culler
+    ${current_timeout} =  Get Culler Timeout
+    Should Be Equal  ${DEFAULT_CULLER_TIMEOUT}  ${current_timeout}
+    Close Browser
+
+Verify Culler Timeout Can Be Updated
+    [Documentation]    Verifies culler timeout can be updated
+    [Tags]    Sanity    Tier1
+    ...       ODS-1231
+    Modify Culler Timeout    ${CUSTOM_CULLER_TIMEOUT}
+    ${current_timeout} =  Get Culler Timeout
+    Should Not Be Equal  ${current_timeout}  ${DEFAULT_CULLER_TIMEOUT}
+    Should Be Equal   ${current_timeout}  ${CUSTOM_CULLER_TIMEOUT}
+    Close Browser
+
+Verify Culler Kills Inactive Server
+    [Documentation]    Verifies that the culler kills an inactive
+    ...    server after timeout has passed.
+    [Tags]    Sanity    Tier1
+    ...       ODS-1254
+    Spawn Minimal Image
+    Clone Git Repository And Run    https://github.com/redhat-rhods-qe/ods-ci-notebooks-main    ods-ci-notebooks-main/notebooks/500__jupyterhub/notebook-culler/Inactive.ipynb
+    Open With JupyterLab Menu    File    Save Notebook
+    Close Browser
+    Sleep    ${${CUSTOM_CULLER_TIMEOUT}+120}
+    ${notebook_pod_name} =  Get User Notebook Pod Name  ${TEST_USER.USERNAME}
+    Run Keyword And Expect Error  Pods not found in search  OpenShiftLibrary.Search Pods
+    ...    ${notebook_pod_name}  namespace=rhods-notebooks
+
+Verify Culler Does Not Kill Active Server
+    [Documentation]    Verifies that the culler does not kill an active
+    ...    server even after timeout has passed.
+    [Tags]    Sanity    Tier1
+    ...       ODS-1253
+    Spawn Minimal Image
+    Clone Git Repository And Open    https://github.com/redhat-rhods-qe/ods-ci-notebooks-main    ods-ci-notebooks-main/notebooks/500__jupyterhub/notebook-culler/Active.ipynb
+    Open With JupyterLab Menu    Run    Run All Cells
+    Open With JupyterLab Menu    File    Save Notebook
+    Close Browser
+    Sleep    ${${CUSTOM_CULLER_TIMEOUT}+120}
+    ${notebook_pod_name} =  Get User Notebook Pod Name  ${TEST_USER.USERNAME}
+    OpenShiftLibrary.Search Pods  ${notebook_pod_name}  namespace=rhods-notebooks
+
+Verify Do Not Stop Idle Notebooks
```

Comment:
hmm what about something like `Verify Culler Does Not Kill Server When Disabled`?